internal/subscription/core.go

```diff
@@ -344,6 +346,57 @@ func createDelayTopics(ctx context.Context, m *Model, topicCore topic.ICore, top
 	return nil
 }
 
+// Get all the subs and rescale all the Retry/Delay/DLQ topics
+func (c *Core) RescaleSubTopics(ctx context.Context, topicModel *topic.Model, partitions int) error {
+	subList, err := c.List(ctx, Prefix+topicModel.ExtractedProjectID)
+	if err != nil {
+		return err
+	}
+	logger.Ctx(ctx).Info("Sublist: ", subList)
+	for index, m := range subList {
+		logger.Ctx(ctx).Info("At Index: ", (index), " | ", m)
+
+		retryModel := &topic.Model{
+			Name:               m.GetRetryTopic(),
+			ExtractedTopicName: m.ExtractedSubscriptionName + topic.RetryTopicSuffix,
+			ExtractedProjectID: m.ExtractedTopicProjectID,
+			NumPartitions:      partitions,
+		}
+		err1 := c.topicCore.UpdateTopic(ctx, retryModel)
+		if err1 != nil {
+			logger.Ctx(ctx).Error(err1.Error())
+		}
+		for _, delayTopic := range m.GetDelayTopics() {
+			delayModel := &topic.Model{
+				Name:               delayTopic,
+				ExtractedTopicName: strings.Split(delayTopic, "/")[3],
```

Comment:
There should be a helper function that already does this.

Reply:
The already existing helper function was performing the operation other way round, adding another one in Topic package helper module